caffe2/contrib/tensorrt/tensorrt_op_trt.cc

```diff
@@ -0,0 +1,224 @@
+#include "caffe2/contrib/tensorrt/tensorrt_op_trt.h"
+#include "caffe2/core/logging.h"
+
+#include <numeric>
+#include <unordered_map>
+
+namespace caffe2 {
+
+namespace {
+// Note that input of trt tensor is in CHW format, while our tensor is NCHW
+// \return -1 if there is dimension mismatch between C2 tensor and trt tensor.
+// Otherwise, return the multiplicaton of CHW dimensions
+int64_t CheckDims(
+    const nvinfer1::Dims& nv_dims,
+    const std::vector<TIndex>& c2_dims) {
+  if (nv_dims.nbDims + 1 != c2_dims.size()) {
+    return -1;
+  }
+  int64_t chw = 1;
+  for (int i = 0; i < nv_dims.nbDims; ++i) {
+    if (nv_dims.d[i] != c2_dims[i + 1]) {
+      return -1;
+    }
+    chw *= nv_dims.d[i];
+  }
+  return chw;
+}
+
+} // namespace
+
+// Upon construction, we build the inference enigne by deserializing from
+// protobuf string. And since we know the input/output blobs, we can do the
+// binding here too.
+TensorRTOp::TensorRTOp(const OperatorDef& operator_def, Workspace* ws)
+    : Operator<CUDAContext>(operator_def, ws),
+      logger_((nvinfer1::ILogger::Severity)(
+          OperatorBase::GetSingleArgument<int>("log_verbosity", 2))),
```

Comment:
Please use `minloglevel` from here: https://github.com/pytorch/pytorch/blob/a2a28c0ef1d9a433972fe72fa5b0b9b850ccfcaf/caffe2/core/logging.cc#L106

Reply:
I need to check `minloglevel` to see if their levels align well.